test/index.coffee

```diff
@@ -0,0 +1,89 @@
+Promise = require 'bluebird'
+{ expect } = require 'chai'
+_ = require 'lodash'
+
+gc = require('../lib/index')
+dockerUtils = require('../lib/docker.coffee')
+
+SKIP_GC_TEST = process.env.SKIP_GC_TEST == '1' || false
+
+promiseToBool = (p) ->
+	p.return(true).catchReturn(false)
+
+pullAsync = (docker, tag) ->
+	new Promise (resolve, reject) ->
+		docker.pull tag, (err, stream) ->
+			return reject(err) if err?
+			stream.on('data', _.noop)
+			stream.once('end', resolve)
+
+dockerUtils.getDocker()
+.then (docker) ->
+	describe 'Garbage collection', ->
+		# This test case is a little weird, it requires that no other images are present on
+		# the system to ensure that the correct one is being removed. Because of this, you
+		# can use the SKIP_GC_TEST env var to inform the test suite not to run this test
+		it 'should remove the LRU image', ->
+			this.timeout(600000)
+			return Promise.resolve() if SKIP_GC_TEST
+			# first pull some images, so we know in which order they are referenced
+
+			# Create the promises with a standard map so that they're done in the correct
+			# order (bluebird doesn't guarantee this)
```

Comment:
It guarantees that the array will be iterated in order attaching the mapping function, what that means is that in a case like this where the entire array has readily available values (direct values, or already resolved promises) then it will be called in order, it's only in cases where you are waiting on promises that it will make a difference, ie it does:
```coffee
(arr, mapFn) ->
	Promise.resolve(arr).then(arr) ->
		arr.map (v, i) ->
			Promise.resolve(v).then (v) ->
				mapFn(v, i, arr)
```

You can test with the following,
* Array of values
```js
Promise.map(_.times(10), console.log.bind(console))
```
* Array of fulfilled promises
```js
Promise.map(_.times(10, (i) -> Promise.return(i)), console.log.bind(console))
```
* Array of unfulfilled promises
```js
Promise.map(_.times(10, (i) -> Promise.delay(_.random(10000, 15000)).return(i)), console.log.bind(console))
```